rotkehlchen/chain/evm/decoding/aura_finance/utils.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from typing import TYPE_CHECKING, Final
+
+from eth_utils import to_checksum_address
+
+from rotkehlchen.assets.asset import Asset, UnderlyingToken
+from rotkehlchen.assets.utils import TokenEncounterInfo, get_or_create_evm_token
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.constants import ONE
+from rotkehlchen.constants.prices import ZERO_PRICE
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.misc import NotERC20Conformant
+from rotkehlchen.globaldb.cache import (
+    globaldb_get_unique_cache_value,
+    globaldb_set_unique_cache_value_at_ts,
+)
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import CacheType, EvmTokenKind, Price
+from rotkehlchen.utils.misc import ts_now
+
+from .constants import AURA_BOOSTER_ABI, CHAIN_ID_TO_BOOSTER_ADDRESSES, CPT_AURA_FINANCE
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.inquirer import Inquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_aura_pools(evm_inquirer: 'EvmNodeInquirer') -> None:
```

Comment:
So this is reusing a lot of code that has been abstracted by nicholas and used in places like curve lend and morpho. Check how it's done there by using `update_cached_vaults` and adjust this function to abstract it properly